hack/ci/check-doc-only-update.sh

```diff
@@ -1,7 +1,14 @@
 #!/usr/bin/env bash
 
-if [ "$TRAVIS_COMMIT_RANGE" != "" ] && ! git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(\.md)|(\.MD)|(\.png)|(\.pdf)|^(doc/)|^(MAINTAINERS)|^(LICENSE)'; then
-  echo "Only doc files were updated, not running the CI."
-  exit
+set -ex
+
+# Make sure the TRAVIS_COMMIT_RANGE is valid, by catching any errors and exiting.
+if [ -z "$TRAVIS_COMMIT_RANGE" ] || ! git rev-list --quiet $TRAVIS_COMMIT_RANGE; then
```

Comment:
This will treat an empty commit range as invalid and fail the CI.

This means the CI will also fail for branches that we push up to the repo with no new commits. From the [travis env var doc](https://docs.travis-ci.com/user/environment-variables/)
>  TRAVIS_COMMIT_RANGE: The range of commits that were included in the push or pull request. (Note that this is empty for builds triggered by the initial commit of a new branch.)

For instance when we push a new release branch `v0.5.x` it has no new commits ahead of the master.
If the commit range is empty for a new branch we still want to run the CI.

Reply:
> For instance when we push a new release branch v0.5.x it has no new commits ahead of the master.

> We can skip the CI as well since there are no new commits. But we shouldn't be failing it.


@hasbro17 fair point, okay, how would you skip it? I haven't found much useful on skipping builds. 🤔

Reply:
Rather than failing or skipping, what if we tested the commit range and:
1. If set and valid, go ahead with the doc check and exit early if only docs changed.
2. If unset, empty string, or invalid, skip the doc check and run the full build.

Reply:
@joelanford I like that approach, any idea if we can do that from the script itself? I think I only found options to do that from the `.travis.yaml` file.

Reply:
This script is being sourced in `.travis.yml`, so my understanding is that calling exit will cause the caller to exit, which is what ends the test with success (`exit 0`) or failure (`exit 1`).

To avoid ending the test, I think we just need to not call `exit`. I think `return 0` or just falling through to the end of the script would work.